### ISSUE TITLE
Unhandled exception when starting RoslyPad

### ISSUE THREAD

**Issue by cocowalla** [aelij/RoslynPad#269]
Just installed RoslynPad 14.1 on Windows 7, and I get:

```
[Main Instruction]
Unhandled Exception

[Content]
System.MissingMethodException: Method not found: 'Void RoslynPad.ExecutionPlatform..ctor(System.String, System.String, System.Collections.Generic.IReadOnlyList`1<RoslynPad.PlatformVersion>, System.Runtime.InteropServices.Architecture, System.String, System.String, Boolean)'.
   at RoslynPad.PlatformsFactory.<GetExecutionPlatforms>d__0.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at System.Collections.Immutable.ImmutableArray.CreateRange[T](IEnumerable`1 items)
   at RoslynPad.UI.OpenDocumentViewModel..ctor(IServiceProvider serviceProvider, MainViewModelBase mainViewModel, ICommandProvider commands, IAppDispatcher appDispatcher, ITelemetryProvider telemetryProvider)
   at lambda_method(Closure , LifetimeContext , CompositionOperation )
   at System.Composition.Hosting.Core.CompositionOperation.Run(LifetimeContext outermostLifetimeContext, CompositeActivator compositionRootActivator)
   at System.Composition.Hosting.Core.LifetimeContext.TryGetExport(CompositionContract contract, Object& export)
   at System.Composition.CompositionContext.GetExport(CompositionContract contract)
   at RoslynPad.UI.SystemCompositionServiceProvider.GetService(Type serviceType)
   at RoslynPad.UI.ServiceProviderExtensions.GetService[T](IServiceProvider serviceProvider)
   at RoslynPad.UI.MainViewModelBase.CreateNewDocument()
   at async RoslynPad.UI.MainViewModelBase.OpenAutoSavedDocuments(?)
   at async RoslynPad.UI.MainViewModelBase.InitializeInternal(?)
   at async RoslynPad.UI.MainViewModelBase.Initialize(?)
```

Contents of `C:\Program Files\dotnet\shared`:

```
- Microsoft.AspNetCore.All
  - ... [gues you don't need these]
- Microsoft.AspNetCore.App
  - ... [gues you don't need these]
- Microsoft.NETCore.App
  - 2.0.3
  - 2.0.5
  - 2.0.6
  - 2.0.7
  - 2.0.9
  - 2.1.0
  - 2.1.1
  - 2.1.2
  - 2.1.3
  - 2.1.4
  - 2.1.5
  - 2.1.6
  - 2.1.7
  - 2.1.8
  - 2.1.9
  - 2.2.1
  - 2.2.2
  - 2.2.3
  - 2.2.4
```

I came across #266 which appears to be the same issue, but I do have .NET Core 2.2 installed already.

**Comment by aelij** [comment]
#266 is not the same issue.

Missing method can only happen if you have older RoslynPad assemblies mixed with newer ones. I suggest you clean up your installation/build folders.

**Comment by cocowalla** [comment]
@aelij this was a totally clean installation from the download on the RoslynPad website, so no old assemblies.

I recently install it on Windows 10, and it works just fine, so presumably some kind of incompatibility with Windows 7? Note I have VS installed and use it as a dev machine, so other dotnet and dotnet core apps work as expected. 

Should this be reopened?

**Comment by aelij** [comment]
@cocowalla I don't see how a `MissingMethodException` for a RoslynPad type could happen on a completely clean install.

I'll keep this closed unless you can provide a full repro.

**Comment by aelij** [comment]
BTW, do you have .NET 4.6.2 installed there? Might be the cause.

**Comment by cocowalla** [comment]
@aelij 

Thanks for that.

I'm not sure what you're expecting here, but the steps to reproduce:

1. Find a Windows 7 machine
2. Install .NET Core
3. Install the latest version of RoslynPad
4. Start RoslynPad

> I don't see how a MissingMethodException for a RoslynPad type could happen on a completely clean install.

Well, it does! I even tried downloading it again, in case the archive was somehow corrupt.

Perhaps there is some kind of dependency (transitive or direct) on a missing Visual C++ Redistributable, or something like that?

**Comment by cocowalla** [comment]
> BTW, do you have .NET 4.6.2 installed there? Might be the cause.

Yes, as well as .NET 4.7.1. I'll install .NET 4.7.2 and see if it makes any difference.

**Comment by cocowalla** [comment]
Yes - installing .NET 4.7.2 has fixed this issue!

Perhaps something in `ExecutionPlatform` relies on a method/class not available before 4.7.2?
